### PR TITLE
fix(Table): handle case where currentPage is greater than total pages

### LINKED ISSUE #1925: Table: onPageChange is getting called on initial load
- `onPageChange` is getting called on initial load.
- [Slack](https://razorpay.slack.com/archives/CMQ3RBHEU/p1704273999167809?thread_ts=1704273978.768959&cid=CMQ3RBHEU)
- [Link](https://stackblitz.com/edit/xui3sa?file=PaginatedTable.tsx)

### PR DESCRIPTION
**Pull request by chaitanyadeorukhkar**
Fixes #1925 

- Fix an edge case where the controlled prop `currentPage` being greater that the total pages makes the table go in an infinite state update. 
- Throw appropriate error
- Add test

### CODE REVIEW

**Review by anuraghazra — APPROVED** [review]

File: packages/blade/src/components/Table/TablePagination.web.tsx [hunk 1]
```diff
@@ -212,7 +213,16 @@ const _TablePagination = ({
   }, [controlledCurrentPage, currentPage, handlePageChange, onPageChange]);
 
   if (currentPage > totalPages - 1) {
-    handlePageChange(totalPages - 1);
+    if (!isUndefined(controlledCurrentPage)) {
```
Comment (saurabhdaware) [inline packages/blade/src/components/Table/TablePagination.web.tsx:216]:
```suggestion
    if (controlledCurrentPage) {
```

this is simpler to read I guess 🤔

Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/Table/TablePagination.web.tsx:216]: controlledCurrentPage can be 0 or 1 which is a valid boolean 😅

**Review by saurabhdaware — APPROVED** [review]

File: packages/blade/src/components/Table/TablePagination.web.tsx (hunk 1 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Table/TablePagination.web.tsx:216]: ohhh

**Review by snitin315 — APPROVED** [review]
:shipit:
